api/system/services/System.js

```diff
@@ -1,8 +1,106 @@
 'use strict';
+const fetch = require('node-fetch');
+
 
 /**
  * Read the documentation (https://strapi.io/documentation/3.0.0-beta.x/guides/services.html#core-services)
  * to customize this service
  */
 
-module.exports = {};
+module.exports = {
+  /**
+   * Promise to add record
+   *
+   * @return {Promise}
+   */
+
+  create: async (values) => {
+    let edsmData = await strapi.services.system.edsmUpdate(values);
+
+    return strapi.query('System').create(edsmData);
+  },
+
+  /**
+   * Promise to edit record
+   *
+   * @return {Promise}
+   */
+
+  update: async (params, values) => {
+
+    if (!values.edsmCoordLocked){
+      let oldData = await strapi.services.system.findOne({id: params.id});
+      oldData = oldData.toJSON();
+
+      if (oldData.edsmCoordLocked == false || oldData.edsmCoordLocked == undefined) {
+        if (values.systemName == undefined) {
+          values.systemName = oldData.systemName;
+        }
+        values.missingSkipCount = oldData.missingSkipCount;
+
+        let edsmData = await strapi.services.system.edsmUpdate(values);
+
+        return strapi.query('System').update(params, edsmData);
+      }
+    }
+
+    return strapi.query('System').update(params, values);
+  },
+
+  /**
+   * Promise to fetch a system from EDSM to update a local record
+   *
+   * @return {Promise}
+   */
+  edsmUpdate: async (values) => {
+    const edsmAPISystem = 'https://www.edsm.net/api-v1/system?showId=1&showCoordinates=1&showPrimaryStar=1';
+
+    const getData = async system => {
+      try {
+        const response = await fetch(edsmAPISystem + '&systemName=' + encodeURIComponent(system));
+        const json = await response.json();
+        return json;
+      } catch (error) {
+        console.log(error);
+      }
+    };
+
+    let newValues = values;
+    let newData = {};
+    if (newValues.missingSkipCount >= 10 ) {
+      newValues.systemName = newValues.systemName.toUpperCase();
+      return newValues;
+    } else {
+      newData = await getData(values.systemName);
+    }
+
+    if (newData != undefined) {
+
+      newValues.systemName = newData.name.toUpperCase();
+
+      if (newData.id) {
```

Comment:
Basically ok in this particular instance, but again, if `newData.id: 0` then this will return `false`.